File: maths/cycle1/fiches-methodes3e/I/Ruby/m0208.rb
#!/usr/bin/env ruby

require_relative "method02"
require_relative "corrective"

class Solution < Method02
  include Corrective

  def answering
    chocolates = 346
    capacity = 16
    caskets = chocolates / capacity
    remainder = chocolates % capacity
    puts "#{chocolates} = #{capacity} x #{caskets} + #{remainder}."
    puts "Le confiseur pourra remplir #{caskets} coffrets."
    print "Il lui restera #{remainder} "
    puts remainder > 1 ? 'chocolats.' : 'chocolat.'
  end
end

s = Solution.new
s.exercise_08
bool = true
bool = s.put_up_method?
if bool then s.put_up_method end
s.put_up_correct_version?
s.answering

File: maths/cycle1/fiches-methodes3e/I/Pascal/m0208.pas
program m0208;

{$mode objfpc}{$H+}

uses
    method02, display;

var
    chocolates, caskets : Word;
    capacity, remainder : Byte;
    bool : Boolean;

begin
    WriteLn('# Exercice 8.');
    WriteLn('# ----------');
    WriteLn('# Un confiseur a pu réaliser 346 chocolats. Il veut les présenter pour la vente');
    WriteLn('# dans des coffrets de 16. Combien pourra-t-il remplir de coffrets ?');
    WriteLn('-----------------------------------------------------------------------------------');

    bool := DisplayMethodWording;
    if bool then PutAMethod();

    bool := PutUpCorrectVersion;
    if not bool then exit;

    chocolates := 346;
    capacity := 16;
    caskets := chocolates div capacity;
    remainder := chocolates mod capacity;
    WriteLn(chocolates, ' = ', capacity, ' x ', caskets, ' + ', remainder, '.');
    WriteLn('Le confiseur pourra remplir ', caskets, ' coffrets.');
    Write('Il lui restera ', remainder);
    if (remainder > 1) then
        WriteLn(' chocolats.')
    else
        WriteLn(' chocolat.');
end.

